internal/planner/mapper/mapper.go

```diff
@@ -966,7 +966,10 @@ func resolveInnerFilterDependencies(
 
 	for key := range source {
 		if key == request.FilterOpAnd || key == request.FilterOpOr {
-			compoundFilter := source[key].([]any)
+			compoundFilter, ok := source[key].([]any)
+			if !ok {
+				continue // filter is nil
```

Comment:
suggestion: I do think `if foo == nil` conveys intent much better than a cast+comment

Reply:
I don't mind the type assertion here as it does 2 things. It ensures it's an array and not nil at the same time.

Reply:
I see that as part of the problem, it shouldn't be checking the type here, and it is not really handling it if it is somehow the wrong type - it is just silently ignoring what could be a serious coding error.